test-suite/output/NumberNotations.out

```diff
@@ -419,7 +419,15 @@ Ip3 (S (S (S O))) nat bool
 eq_refl
      : 1 = 1
 0
+     : 0 = 0
+eq_refl
+     : id 0 = id 0
+eq_refl
      : 1 = 1
+0
+     : 0 = 0
+eq_refl
```

Comment:
This should print as `0 : id 0 = id 0`.  Is there a way to see what the typing error is?

Reply:
It bails because of allow_constant:false at https://github.com/coq/coq/blob/93f0cc96dbb825c467759aee5573e6969a3d9a0c/interp/notation.ml#L682

Reply:
Ah, I see.  In that case, before
```coq
Number Notation eqO eqO_of_uint eqO_to_uint : nat_scope.
```
could you add
```coq
Notation eqS := (eq _ (S _)) (only parsing).
Number Notation eqS eqO_of_uint eqO_to_uint : nat_scope.

Check 42.
Check eq_refl (S O).  (* matches eqS, but printer type errors, so not printed *)
Check eq_refl O. (* does not match eqS, printer not called *)
Check eq_refl (id O). (* contains a global constant, printer not called *)
```
to test that when the notation doesn't match we don't print

Reply:
I get `Cannot infer this placeholder of type "nat".` on `Number Notation eqS ...`

Reply:
Uh, this seems suspicious, but I guess it doesn't have to be solved in this PR.  Why is `(eq _ _)` fine but `(eq _ (S _))` not?  / Where does this error come from?

Reply:
https://github.com/coq/coq/blob/0b1ae026456fe950c895bfe5ecc519402dcfa4b0/plugins/syntax/number.ml#L403
`eq _ _` falls in the `NHole` case a couple lines above

Reply:
This looks to me like a limitation of the current implem.
This shouldn't prevent merging the current PR.